tests/yam/agama/generate_dud.pm

```diff
@@ -26,6 +26,12 @@ sub run {
     assert_script_run("curl -o tmp/dud/root/autoinst.json $profile_url");
     assert_script_run("mkdud --create $dud tmp/dud/root --dist tw");
     upload_asset($dud);
+
+    # https://progress.opensuse.org/issues/185713
+    my $arch = get_required_var('ARCH');
+    $dud = "kernel-" . $arch . ".dud";
```

Comment:
let's rename `$dud` to `$archived_dud_with_profile` and create new one for this with `$archived_dud_with_kernel_module` for clarity.
the not-archived one, simple rpm we don't need to generate it of course but I think it helps to understand.